de.dlr.sc.virsat.graphiti.test/src/de/dlr/sc/virsat/graphiti/diagram/BeanIndependenceSolverTest.java

```diff
@@ -76,91 +86,108 @@ protected void doExecute() {
-		assertEquals(true, permission);
-		
-		Resource resEd = rs.getStructuralElementInstanceResource(ed.getStructuralElementInstance());
-		resEd.getContents().add(ed.getStructuralElementInstance());
+		getAndAddStructuralElementInstance(sei1);
+		checkDiagramWritePermissionForStructuralElement(sei1);
+
+		addStructuralElementToResource(ed1);
 		Object obj = beanIndependenceSolver.getBusinessObjectForKey(seiKey);
 		aBean = (ABeanStructuralElementInstance) obj;
-		assertEquals(ed, aBean);
+		assertEquals(ed1, aBean);
+	}
+
+	@Test
+	public void testSolveIndependenceForBeanProperty() throws CoreException, IOException {
+		Object beanNoteObj = beanIndependenceSolver.getBusinessObjectForKey(testDocument2.getNoteBean().getUuid());
+		assertEquals(testDocument2.getNoteBean(), beanNoteObj);
+	}
+
+	@Test
+	public void testSolveIndependenceForCa() throws CoreException, IOException {
+		String caKey = beanIndependenceSolver.getKeyForBusinessObject(testDocument);
+		assertEquals(DOCUMENTUUID, caKey);
 
 		Object documentObj = beanIndependenceSolver.getBusinessObjectForKey(testDocument2.getUuid());
-		ABeanCategoryAssignment aCategory = (ABeanCategoryAssignment) documentObj;
-		assertEquals(testDocument2, aCategory);
-		
+		assertEquals(testDocument2, documentObj);
+	}
+
+	@Test
+	public void testSolveIndependenceNull() throws CoreException, IOException {
 		assertEquals(null, beanIndependenceSolver.getKeyForBusinessObject(null));
+	}
+
+	private void checkDiagramWritePermissionForStructuralElement(StructuralElementInstance sei) {
+		Boolean permission = DiagramHelper.hasDiagramWritePermission(sei);
+		assertEquals(true, permission);
+	}
 
+	private void addStructuralElementToResource(ElementDefinition ed) {
+		Resource resEd2 = rs.getStructuralElementInstanceResource(ed.getStructuralElementInstance());
+		resEd2.getContents().add(ed.getStructuralElementInstance());
+	}
+
+	/**
+	 * @return IDiagramTypeProvider
+	 */
+	private IDiagramTypeProvider getDiagramTypeProvider() {
+		// Create The diagram
+		Diagram diagram = Graphiti.getPeCreateService().createDiagram("test", "testDiagram", true);
+		IFolder diagramFolder = testProject.getFolder("src/diagrams/");
+		IFile diagramFile = diagramFolder.getFile("testDiagram" + "." + "test");
+		URI uri = URI.createPlatformResourceURI(diagramFile.getFullPath().toString(), true);
+		DiagramHelper.createDiagram(uri, diagram, rs);
+
+		IDiagramTypeProvider dtp = GraphitiInternal.getEmfService().getDTPForDiagram(diagram);
+		GraphitiInternal.getEmfService().wireDTPToDiagram(diagram, dtp);
+		dtp.init(diagram, null);
+		return dtp;
+	}
+
+	/**
+	 * @param sei StructuralElementInstance
+	 */
+	private void getAndAddStructuralElementInstance(StructuralElementInstance sei) {
+		editingDomain.getVirSatCommandStack().execute(new RecordingCommand(editingDomain) {
+			@Override
+			protected void doExecute() {
+				rs.getAndAddStructuralElementInstanceResource(sei);
+			}
+		});

```

Comment:
The calls to the command stack can be simplified by a new method from the super class

```suggestion
		executeAsCommand(() -> rs.getAndAddStructuralElementInstanceResource(sei));
```